include/libkernelflinger/vars.h

```diff
@@ -127,7 +127,8 @@ EFI_STATUS set_slot_fallback(BOOLEAN enabled);
 enum device_state {
 	UNKNOWN_STATE = -1,
 	LOCKED = 0,
-	UNLOCKED = 1
+	UNLOCKED = 1,
+	JUST_INIT = 2
```

Comment:
This does not make sense.  This is no a device_state.  You should have add a parameter to your `read_XXX_state()` function to get the information you need for initialization purposes.